tests/handlers/tests.py

```diff
@@ -335,11 +335,13 @@ def test_root_path(self):
         self.assertEqual(request.script_name, "/root")
         self.assertEqual(request.path_info, "/somepath/")
 
-    @override_settings(FORCE_SCRIPT_NAME="/FORCED_PREFIX/")
+    @override_settings(FORCE_SCRIPT_NAME="/FORCED_PREFIX")
```

Comment:
I _think_ `SCRIPT_NAME` and `FORCE_SCRIPT_NAME` _shouldn't_ end in a slash but we handle this in WSGI after the changes in this old [ticket 20169](https://code.djangoproject.com/ticket/20169). I think `path_info` _should_ start with a slash or be an empty string ([feel like this is implied here](https://wsgi.readthedocs.io/en/latest/definitions.html#envvar-PATH_INFO)).

After these changes in ASGI, if you have `FORCE_SCRIPT_NAME` end in a slash `path_info` will not start with a slash. I don't think this is a problem?

I am more worried about whether the 'to start/end in slash, not to start/end in a slash' is important and if so, documented appropriately. Specifically:
- the docs for [`FORCE_SCRIPT_NAME`](https://docs.djangoproject.com/en/5.0/ref/settings/#force-script-name) to me imply the default of `SCRIPT_NAME` is `/` whereas I think [WSGI implies it's an empty string](https://wsgi.readthedocs.io/en/latest/definitions.html#envvar-SCRIPT_NAME). 
- we also set `SCRIPT_NAME` with an ending slash in the [Making requests testing topic](https://docs.djangoproject.com/en/5.0/topics/testing/tools/#making-requests)

Do you think we need to handle the trailing slash in ASGI the same as we do for WSGI? 
Do we need a test to clarify this for ASGI (similar to  `requests_tests.tests.RequestsTests.test_wsgirequest_path_with_force_script_name_trailing_slash`)?
Do you think we need a docs update? This might be a separate clean-up PR.

Maybe none of this matters very much and I'm very happy to follow your guidance here 👍

Reply:
So, I'm pretty clear that in ASGI at least, `root_path` is meant not to have the trailing slash. 

https://github.com/django/asgiref/pull/446/files -- e.g. see test case. 

(Starlette and FastAPI do the same, as do Channels and Daphne.)

I removed the trailing slash here as "the tests are docs" so we at least show the expected use-case. 

- I'm not 100% sure about the exact rules in WSGI and whether we need to bulletproof various possibilities. (My suspicion would be to let a report come in before spending too long on possible bugs...)

Reply:
IDK 🤯 :)

I think this is right: 

> I think SCRIPT_NAME and FORCE_SCRIPT_NAME shouldn't end in a slash but we handle this in WSGI after the changes in this old ticket 20169. I think path_info should start with a slash or be an empty string (feel like this is implied here).

I wouldn't add extra handling on the ASGI side here immediately: folks should just do it right. 

I'm not sure if there's a need to *tighten* any current behaviour. (It's pretty niche…) 

If you have _ganas_ to add extra tests, please don't let me talk you out of it! 😉

Reply:
I've been (also?) confused by the `FORCE_SCRIPT_NAME` setting. There have been multiple reports about related issues, such as ticket-7930, ticket-24967, ticket-31724, and dupes.

I think the common denominator is lack of understanding on the correct format for the value to use in the setting ("to slash or not to slash" :tophat: :guitar:), so I would appreciate if our docs could be more assertive in this regard (not that this belong to this PR, but I think it was relevant to the conversation).